fix for , non presence of consul-version meta

vijayraghav-io opened the pull request:
Fixes #18406 

Hi @david-yu 
Fixed the ui for non-presence of consul-version in Node meta.

natmegs commented on ui/packages/consul-ui/app/serializers/application.js:
```diff
@@ -227,36 +227,38 @@ export default class ApplicationSerializer extends Serializer {
     let versionSet = new Set();
     payload.forEach(function (item) {
-      if (item.Meta && item.Meta['consul-version'] !== '') {
+      if (item.Meta && item.Meta['consul-version'] && item.Meta['consul-version'] !== '') {
```
```suggestion
      if (item.Meta && item.Meta['consul-version']) {
```

Checking for truthiness here should be sufficient :)

vijayraghav-io replied:
done

natmegs commented on ui/packages/consul-ui/app/serializers/application.js:
```diff
@@ -227,36 +227,38 @@ export default class ApplicationSerializer extends Serializer {
-        const number = Number(part);
-        return isNaN(number) ? 0 : number;
-      });
+    if (versionArray.length > 0) {
```
I think we can remove this as sort will be a noop if the array is empty

vijayraghav-io replied:
@natmegs Yes! , was extra cautious :) , will update.

natmegs reviewed (commented):
Hey @vijayraghav-io thanks so much for this contribution!! This is working great and we really appreciate this bug fix

It would be great to add a unit test for this case and the `getDistinctConsulVersions` serializer method in the [application serializer unit tests](https://github.com/hashicorp/consul/blob/5fb9df1640cfa1aaf96228da1f44b31ecd6b825a/ui/packages/consul-ui/tests/unit/serializers/application-test.js#L10) if possible.

It would also be awesome to update our [mocks](https://github.com/hashicorp/consul/blob/2f20c77e4df9d663ee5f06d811c23dd9ad14db88/ui/packages/consul-ui/mock-api/v1/internal/ui/nodes#L28-L29) so that if agentless is enabled and it is a synthetic node, the consul version is empty. This would help to more closely mimic the behaviour of the real API and would be super helpful for dev and testing :)

vijayraghav-io commented:
@natmegs @david-yu 
done the updates, also updated mock-api for synthetic-node cases. 
And added test for getDistinctConsulVersions serializer method

david-yu commented:
Thanks, I've added a backport. If you're able to create a changelog in the .changelog folder that would be amazing.

vijayraghav-io commented:
added change log.

hc-github-team-consul-core commented:
Backport failed @WenInCode. Run: https://github.com/hashicorp/consul/actions/runs/5944665998

david-yu commented:
Re-triggering backport.
